src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/ProjectProperties.cs

```diff
@@ -12,10 +12,8 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore
     internal class ProjectProperties : ImmutablePropertyCollection<IVsProjectProperty>, IVsProjectProperties
     {
         public ProjectProperties(IEnumerable<IVsProjectProperty> items)
-            : base(items)
+            : base(items, item => item.Name)
```

Comment:
This is cached once, right?

Reply:
The removal of the virtual also avoids capture class & persumably caches the delegate.

Reply:
Appears to be, yes.

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAcoKgW1QOzgBsiBjACwlIGsBLfAcwB8ABAJgEYBYAKF5YDMAAnZCAynGBCQQgEJRkEXgG9eQkcInAAFAEppQ4AojbkQgLwA+Icn3KAvr0d8egkWznGVaoegQB7GCoggBNPRW0WAFYAHhYOAAYAGniE6zA4fFI7Z3sgA